.pre-commit-config.yaml

```diff
@@ -154,61 +154,21 @@ repos:
       # NOTE: CI automations run both checks. Whenever the versions are
       # NOTE: updated in this config, a symmetric action must be performed
       # NOTE: in `tox.ini` as well.
-      - &_base-mypy-setup-map
-        id: mypy
+      - id: mypy
         alias: mypy-py310
         name: MyPy, for Python 3.10
-        additional_dependencies:
-          - lxml == 4.6.4 # needed for `--txt-report`
-          - types-docutils
+        additional_dependencies: &_mypy_deps
           - types-PyYAML
+          - types-dataclasses # Needed for checking py3.6 with a later interpreter
+          - types-docutils
         args:
-          # FIXME: get rid of missing imports ignore
-          - --ignore-missing-imports
-          - --install-types
-          - --namespace-packages
-          - --non-interactive
-          - --pretty
           - --python-version=3.10
-          - --show-column-numbers
-          - --show-error-codes
-          - --show-error-context
-          #- --strict
-          #- --strict-optional
-          - --txt-report
-          - .tox/.tmp/.mypy/
-          - docs/
-          - share/
-          - src/
-          - tests/
         pass_filenames: false
-      - <<: *_base-mypy-setup-map
-        alias: mypy-py36
+      - id: mypy
```

Comment:
This still needs an alias.

Reply:
How come?

Reply:
This would allow being explicit about choosing what to test against.

Reply:
Not sure I understand, there is no, and should be no configuration in the pre-commit related to the "what" now that the mypy.ini file exists.  Looking at it, I don't see that is does anything in this case.  Is it future proofing for something that might happen later?

Reply:
This allows doing `tox -e lint -- mypy-py36 --all-files` and not `tox -e lint -- mypy{an_unknown_and_surprising_unobvious_version_that} --all-files`

Reply:
Still not sure I understand, with the following configuration

```
      - id: mypy
        alias: mypy-py310
        name: MyPy, for Python 3.10
        additional_dependencies:
          - lxml == 4.6.4 # needed for `--txt-report`
          - types-docutils
          - types-PyYAML
        args:
          - --python-version=3.10
        pass_filenames: false
      - id: mypy
        alias: mypy-py36
        name: MyPy, for Python 3.6
        additional_dependencies:
          - lxml == 4.6.4 # needed for `--txt-report`
          - types-dataclasses # Needed for checking py3.6 with a later interpreter
          - types-docutils
          - types-PyYAML
        args:
          - --python-version=3.6
```
Your first example works fine:

```
(venv) x1 ➜  ansible-navigator git:(mypy_1) ✗ tox -qe lint -- mypy-py36 --all-files
MyPy, for Python 3.6.....................................................Passed
```

and your second fails as expected

```
No hook with id `mypy-py35` in stage `commit`
```

Reply:
It was to show that barebones `mypy` id would not make it obvious for the invoker what they run. Plus, using just `mypy` would match two hooks py3.10 and py3.6, not one. Aliases add additional id but they don't prevent the original one from being used.
Just try `tox -e lint -- mypy --all-files` and you'll see that it doesn't let you specify the second check only.

Reply:
that said.................. I've added it back since it is unrelated to the move of the configuration.

I do that in a separate PR were is doesn't distract from moving the configuration.

Reply:
FWIW I think that my previous comment demonstrates that it shouldn't be changed, even in a separate PR.

Reply:
The reason it shouldn't be changed is `pass_filenames` would need to be repeated.

```
[{'id': 'mypy', 'alias': 'mypy-py310', 'name': 'MyPy, for Python 3.10', 'additional_dependencies': ['lxml == 4.6.4', 'types-docutils', 'types-PyYAML'], 'args': ['--python-version=3.10'], 'pass_filenames': False}, {'id': 'mypy', 'alias': 'mypy-py36', 'name': 'MyPy, for Python 3.6', 'additional_dependencies': ['lxml == 4.6.4', 'types-dataclasses', 'types-docutils', 'types-PyYAML'], 'args': ['--python-version=3.6'], 'pass_filenames': False}]
[{'id': 'mypy', 'alias': 'mypy-py310', 'name': 'MyPy, for Python 3.10', 'additional_dependencies': ['lxml == 4.6.4', 'types-docutils', 'types-PyYAML'], 'args': ['--python-version=3.10'], 'pass_filenames': False}, {'id': 'mypy', 'alias': 'mypy-py36', 'name': 'MyPy, for Python 3.6', 'additional_dependencies': ['lxml == 4.6.4', 'types-dataclasses', 'types-docutils', 'types-PyYAML'], 'args': ['--python-version=3.6']}]
```

Reply:
If it were, the deserialized yaml would be the same.  By having it, it is more likely the 2 will share a common configuration and something won't get missed.

Reply:
Oh, I see now! You were talking about YAML anchors and I was talking about the pre-commit setting called `alias`. Is that right? I only wanted that alias to be preserved. But I also don't see a problem with YAML aliases — the mapping values can be overridden.

Reply:
lol.................................... :)  I should have asked which alias :)

Reply:
Yeah, I should've commented on the removed line with the setting too. Now I see how this was confusing that this discussion is attached to the `id` option line...